x-pack/plugins/ml/public/application/data_frame_analytics/common/use_results_view_config.ts

```diff
@@ -83,12 +83,12 @@ export const useResultsViewConfig = (jobId: string) => {
               setIsLoadingJobConfig(false);
             }
           } catch (e) {
-            setJobCapsServiceErrorMessage(getErrorMessage(e));
+            setJobCapsServiceErrorMessage(extractErrorMessage(e));
```

Comment:
Noticed while testing this section that errors never get into this block when the error occurs in `initializeFromIndexPattern` since that function catches any error and logs it and does not propagate it. This means the results view table just fails silently and is empty. Maybe outside the scope of this PR but would be good to talk about whether the errors should be propagated somehow to code calling the service.

Reply:
yes the errors probably should be propagated. `initializeFromIndexPattern` is called a few places and it's only here that we're expecting to catch anything.
I agree that it's outside of the scope of this PR, but definitely worth revisiting.